### PR TITLE
Compute shader JSDocs + small API changes

### PR DESCRIPTION
**Pull request by mvaligursky**
- feature: compute shader can use multiple uniform buffers, instead of just a single default one
- all API needed to use Compute shader is now public, from Uniform buffer / Bind group formats to StorageBuffer and device submission.

### CODE REVIEW

File: src/platform/graphics/bind-group-format.js [hunk 1]
```diff
@@ -72,9 +145,28 @@ class BindTextureFormat extends BindBaseFormat {
 }
 
 /**
- * @ignore
+ * A class to describe the format of the storage texture for. Storage texture is a texture created
```
Comment (willeastcott) [inline src/platform/graphics/bind-group-format.js:148]: For? For what? 😄

File: src/platform/graphics/bind-group-format.js [hunk 2]
```diff
@@ -72,9 +145,28 @@ class BindTextureFormat extends BindBaseFormat {
...
+ * with the storage flag set to true, which allows it to be used as an output of a compute shader.
+ * Note: At the current time, storage textures are only supported in compute shaders, in a
```
Comment (willeastcott) [inline src/platform/graphics/bind-group-format.js:150]:
```suggestion
 * Note: At the current time, storage textures are only supported in compute shaders in a
```

File: src/platform/graphics/bind-group-format.js [hunk 3]
```diff
@@ -87,26 +179,48 @@ class BindStorageTextureFormat extends BindBaseFormat {
...
     storageBufferFormats = [];
 
     /**
+     * Create a new instance.
+     *
      * @param {import('./graphics-device.js').GraphicsDevice} graphicsDevice - The graphics device
      * used to manage this vertex format.
-     * @param {(BindBaseFormat|BindTextureFormat|BindStorageTextureFormat|BindStorageBufferFormat)[]} formats
-     * - An array of bind formats.
+     * @param {(BindTextureFormat|BindStorageTextureFormat|BindUniformBufferFormat|BindStorageBufferFormat)[]} formats
```
Comment (willeastcott) [inline src/platform/graphics/bind-group-format.js:220]: Could this just be `BindBaseFormat[]`?

File: src/platform/graphics/storage-buffer.js [hunk 4]
```diff
@@ -58,14 +58,45 @@ class StorageBuffer {
         vram.sb += size;
     }
 
+    /**
+     * Read a content of a storage buffer.
```
Comment (willeastcott) [inline src/platform/graphics/storage-buffer.js:62]:
```suggestion
     * Read the contents of a storage buffer.
```

File: src/platform/graphics/storage-buffer.js [hunk 5]
```diff
@@ -58,14 +58,45 @@ class StorageBuffer {
...
+     *
+     * @param {number} [offset] - The byte offset of data to read. Defaults to 0.
+     * @param {number} [size] - The byte size of data to read. Defaults to the full size of the
+     * buffer minus the offset.
+     * @param {ArrayBufferView|null} [data] - Typed array to populate with the data read from the storage
```
Comment (willeastcott) [inline src/platform/graphics/storage-buffer.js:67]:
```suggestion
     * @param {ArrayBufferView|null} [data] - Typed array to populate with the data read from the storage.
```

File: src/platform/graphics/bind-group-format.js (hunk 3 above)
Comment (mvaligursky) [reply src/platform/graphics/bind-group-format.js:220]: yeah but I want to avoid people thinking they can specify an instance of the BindBaseFormat too.

**Pull request merged**